csrc/mmdeploy/codebase/mmseg/segment.cpp

```diff
@@ -31,10 +32,15 @@ class ResizeMask : public MMSegmentation {
     auto mask = inference_result["output"].get<Tensor>();
     MMDEPLOY_DEBUG("tensor.name: {}, tensor.shape: {}, tensor.data_type: {}", mask.name(),
                    mask.shape(), mask.data_type());
-    if (!(mask.shape().size() == 4 && mask.shape(0) == 1 && mask.shape(1) == 1)) {
+    if (!(mask.shape().size() == 4 && mask.shape(0) == 1)) {
```

Comment:
so here do not support batch > 1? this is not good for tensorrt where multi-batch could be used.

Reply:
This is not introduced by this PR.

Reply:
@lvhan028 Hi, not so sure about this. Does this mean no multiple batch inference is supported in skd for mmseg?

Reply:
Hi @RunningLeon I'm also very interested is this - Is batch inference possible? (n_samples > 1). 
For example, I have a script for test time augmentation which converts the image into 8 different images, and I want to process them all in one batch for inference. I am currently using:
![image](https://user-images.githubusercontent.com/46079516/204425094-554d41e5-2fa9-41f6-af45-4b59a012c7f7.png)

Then I want to move this inference pipeline onto AWS for large scale GPU accelerated TRT batch inference.

Also, will TensorRT Segmentation probabilities (instead of class labels) be possible in the near future?
Thanks, Sam

Reply:
> 

In SDK, batch inference tensors are carefully handled by `NetModule`. They will be sliced into single ones.
Refer to https://github.com/open-mmlab/mmdeploy/blob/b521e7da03c04fa5c783cb881f0f249f2cba9bb9/csrc/mmdeploy/net/net_module.cpp#L174
Thus, the postprocessing module just needs to focus on the `batch=1` case

Reply:
@lvhan028 Ok thanks, so Python API and model converter cannot do batch inference, but SDK can?

![image](https://user-images.githubusercontent.com/46079516/204433936-da1e1647-7f64-4c32-b09b-b757950665ee.png)

Reply:
@lvhan028 is SDK the only way to do batch inference for Semantic Segmentation in MMDeploy?

Also, is batch inference still possible if I generated the TRT engine model using?:
![image](https://user-images.githubusercontent.com/46079516/205205624-d5334e5f-6b40-4fc8-8088-582d00e8a1d4.png)

Thanks, Sam